service/src/main.rs

```diff
@@ -154,166 +231,237 @@ impl Inclusion for InclusionServiceArc {
-            prover_network_job_id.to_vec()
-        } else {
-            debug!("Preparing request to Celestia...");
-            let height = job.height;
+    /// Connect to the Cestia [Client] and attempt to get a NMP for a [Job].
+    /// A successful Result indicates that the queue DB contains valid ZKP input
+    async fn get_zk_proof_input_from_da(
+        &self,
+        job: &Job,
+        job_key: Vec<u8>,
+        client: Arc<Client>,
+    ) -> Result<(), InclusionServiceError> {
+        debug!("Preparing request to Celestia...");
+        let height = job.height;
+
+        debug!("Getting blob from Celestia...");
+        let try_blob = client
+            .blob_get(job.height.into(), job.namespace, job.commitment)
+            .await;
+        // TODO this is a messy way to handle edge case errors...
+        if let Err(e) = &try_blob {
+            error!("Celestia Client error: {e}");
+            match e {
+                ClientError::Call(error_object) => {
+                    // TODO: make this handle errors much better! JSON stringyness is a problem!
+                    if error_object.message() == "header: not found" {
+                        let e = InclusionServiceError::CelestiaError("header: not found. Likely Celestia Node is missing blob, although it exists on the network overall.".to_string());
+                        error!("{job:?} failed, recoverable: {e}");
+                        let job_status =
+                            JobStatus::Failed(e, Some(JobStatus::DataAvalibilityPending.into()));
+                        return self.finalize_job(&job_key, job_status);
+                    };
+                    if error_object.message() == "blob: not found" {
+                        let e = InclusionServiceError::CelestiaError("blob: not found".to_string());
+                        error!("{job:?} failed, not recoverable: {e}");
+                        let job_status = JobStatus::Failed(e, None);
+                        return self.finalize_job(&job_key, job_status);
+                    };
+
+                    let e =
+                        InclusionServiceError::CelestiaError("UNKNOWN client error".to_string());
+                    error!("{job:?} failed, not recoverable: {e}");
+                    let job_status = JobStatus::Failed(e, None);
+                    return self.finalize_job(&job_key, job_status);
+                }
+                // TODO: handle other Celestia JSON RPC errors
+                _ => {
+                    let e = InclusionServiceError::CelestiaError(
+                        "Unhandled Celestia SDK error".to_string(),
+                    );
+                    error!("{job:?} failed, not recoverable: {e}");
+                    let job_status = JobStatus::Failed(e, None);
+                    return self.finalize_job(&job_key, job_status);
+                }
+            }
+        };
+        let blob = try_blob.expect("exhaustive error handling");
```

Comment:
I really don't  like how verbose this is. For sure want to followup with something more clean & flexible.